File: 2924. Find Champion II.py
class Solution(object):
    def findChampion(self, n, edges):
        in_deg = [0] * n
        for edge in edges:
            in_deg[edge[1]] += 1
        answer, count = -1, 0
        for idx, i in enumerate(in_deg):
            if i == 0:
                answer = idx
                count += 1
        if count == 1:
            return answer
        else:
            return -1

File: java/2924. Find Champion II.java
class Solution {
    public int findChampion(int n, int[][] edges) {
        boolean[] childNodes = new boolean[n];
        int count = 0;
        for (int[] e : edges) {
            if (!childNodes[e[1]]) count++;
            childNodes[e[1]] = true;
        }
        if (count != n-1) return -1;
        for (int i = 0; i < n; i++) {
            if (childNodes[i]) continue;
            return i;
        }
        return -1;
    }
}
